hazelcast/src/main/java/com/hazelcast/internal/util/ExceptionUtil.java

```diff
@@ -44,7 +44,7 @@
      * Interface used by rethrow/peel to wrap the peeled exception
      */
     public interface RuntimeExceptionFactory {
-        RuntimeException create(Throwable throwable, String message);
+        Throwable create(Throwable throwable, String message);
```

Comment:
Broadening `RuntimeExceptionFactory#create` to return `Throwable` may mislead future implementations. Instead we can keep its signature as `RuntimeException create(Throwable t, String m)` and change the new usage in `peelAsNeeded` to do a `sneakyThrow(throwable)`. WDYT?

Reply:
I could not follow. We don't need a throw. We will write the exception to the socket. Not sure how sneaky how will help me here?
I may rename `RuntimeExceptionFactory`  to something like `ExceptionWrapper` which actually what it does. So that the name of the class does not clash with the return type. How does that sound?